src/options.ts

```diff
@@ -0,0 +1,564 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type {
+  ArrayType,
+  Constructor,
+  Data,
+  DataArray,
+  Datum,
+  FieldNames,
+  index,
+  Row,
+  Series,
+  TypedArray,
+  TypedArrayConstructor,
+  Value,
+  ValueArray
+} from "./data.js";
+import {parse as isoParse} from "isoformat";
+import {color, descending, quantile} from "d3";
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray
+const TypedArray = Object.getPrototypeOf(Uint8Array);
+const objectToString = Object.prototype.toString;
+
+export function valueof<T extends undefined | null>(d: T, v: Accessor<T, any>, t?: ArrayType): T;
```

Comment:
In my testing, it seems that when VSCode gives type hints for a function with multiple overrides, it shows them in the order specified in the code. Is there anything else that should affect the order of these overrides? If not, I'd suggest ordering them based on how commonly we expect them to be used.

Reply:
Some of them are more precise and must be before a less precise version that would catch the more precise case. So if we want to reorder them, we'll have to be extra careful.